modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/SettingsImpl.kt

```diff
@@ -194,6 +194,12 @@ class SettingsImpl @Inject constructor(
         sharedPrefs = sharedPreferences,
     )
 
+    override val cacheEntirePlayingEpisode = UserSetting.BoolPref(
+        sharedPrefKey = "cacheEntirePlayingEpisode",
+        defaultValue = false,
+        sharedPrefs = sharedPreferences,
+    )
```

Comment:
It will entirely depend on a PR that depends on this setting but I think it would be safer to have a custom adapter for this setting that always returns `false` if the feature flag is disabled. Otherwise we might get into a situation where users enable the flag, we disable the feature, and they can't come back to the default state because option is not available in the settings.

Reply:
That's a good point. I updated it in [bc9450c](https://github.com/Automattic/pocket-casts-android/pull/2378/commits/bc9450c2fb38cef4cbcc9f305e6cc0dcdbe5c1f6).